rubi/logging_config.py

```diff
@@ -0,0 +1,21 @@
+import logging
+from io import StringIO
+
+# Create an in-memory stream to store logs
+log_stream = StringIO()
+
+# Configure logging to use the in-memory stream
+log_handler = logging.StreamHandler(log_stream)
+log_handler.setLevel(logging.DEBUG)
+formatter = logging.Formatter("%(asctime)s - %(name)s - %(levelname)s - %(message)s")
+log_handler.setFormatter(formatter)
+
+# Configure logging for the root logger
+root_logger = logging.getLogger()
+root_logger.setLevel(logging.DEBUG)
+root_logger.addHandler(log_handler)
+
+
+# Function to retrieve the captured logs
+def get_logs():
```

Comment:
i wonder if we should do something like this here to clear out the logs: 

`def get_logs():`
`    logs = log_stream.getvalue().strip().split("\n")`
`    log_stream.seek(0)`
`    log_stream.truncate()`
`    return logs`

Reply:
discussed offline, there may be more elegant ways to do this that doesn't clear memory in case we want to access those logs later

Reply:
I've removed `get_logs`. Instead, the log behavior can now be edited through the `app.py`. Currently, when new logs are detected, the `logs` list is extended with `new_logs`. I don't think this prints them twice. But this behavior can be customized